### PR TITLE
Homogenize variable explorer UI

### PR DESCRIPTION
**Pull request by goanpeca**
## Description

The "toolbar"  is now horizontal and the options button is aligned to the right, like other plugins.
Also now the variable explorer margins are updated when selection this option for panes inside the preferences

### Before

before-varex

### After
after-varex

## Todo
- [x] Make buttons horizontal
- [x] Align options menu to the right

### CODE REVIEW

File: spyderlib/plugins/variableexplorer.py [hunk 1]
```diff
@@ -126,20 +137,20 @@ def remove_shellwidget(self, shellwidget_id):
...
+            self.stack.setCurrentWidget(nsb)
             if self.isvisible:
                 nsb.visibility_changed(True)
             
     def import_data(self, fname):
         """Import data in current namespace"""
-        if self.count():
-            nsb = self.currentWidget()
+        if self.stack.count():
+            nsb = self.stack.currentWidget()
```
Comment (ccordoba12) [inline spyderlib/plugins/variableexplorer.py]: Please use here `self.current_widget()` instead. I think it's easier to read :-)

File: spyderlib/plugins/variableexplorer.py [hunk 2]
```diff
@@ -151,7 +162,7 @@ def visibility_changed(self, enable):
         """DockWidget visibility has changed"""
         SpyderPluginMixin.visibility_changed(self, enable)
         for nsb in list(self.shellwidgets.values()):
-            nsb.visibility_changed(enable and nsb is self.currentWidget())
+            nsb.visibility_changed(enable and nsb is self.stack.currentWidget())
```
Comment (ccordoba12) [inline spyderlib/plugins/variableexplorer.py]: `self.current_widget()` here too please

File: spyderlib/plugins/variableexplorer.py [hunk 3]
```diff
@@ -167,7 +178,7 @@ def get_focus_widget(self):
         Return the widget to give focus to when
         this plugin's dockwidget is raised on top-level
         """
-        return self.currentWidget()
+        return self.stack.currentWidget()
```
Comment (ccordoba12) [inline spyderlib/plugins/variableexplorer.py]: `self.current_widget()`

File: spyderlib/plugins/variableexplorer.py [hunk 4]
```diff
@@ -126,20 +137,20 @@ def remove_shellwidget(self, shellwidget_id):
...
             nsb = self.shellwidgets.pop(shellwidget_id)
-            self.removeWidget(nsb)
+            self.stack.removeWidget(nsb)
             nsb.close()
     
     def set_shellwidget_from_id(self, shellwidget_id):
         if shellwidget_id in self.shellwidgets:
             nsb = self.shellwidgets[shellwidget_id]
-            self.setCurrentWidget(nsb)
+            self.stack.setCurrentWidget(nsb)
```
Comment (goanpeca) [inline spyderlib/plugins/variableexplorer.py]: Should I make accesors for the places where `stack` is called... so that all the calls are made to the variable explorer object and inside I make the correct call?

Like I did with `self.current_widget`?

Comment (ccordoba12) [inline spyderlib/plugins/variableexplorer.py]: It'd be clearer, yes.

File: spyderlib/widgets/externalshell/namespacebrowser.py [hunk 5]
```diff
@@ -437,8 +437,7 @@ def collapse(self):
         """Collapse"""
         self.sig_collapse.emit()
 
-    @Slot(list)
```
Comment (ccordoba12) [inline spyderlib/widgets/externalshell/namespacebrowser.py]: Why did you change this?

Comment (goanpeca) [inline spyderlib/widgets/externalshell/namespacebrowser.py]: It was producing an error... when clicking on the import button... it must have been sleeping for a while

Comment (ccordoba12) [inline spyderlib/widgets/externalshell/namespacebrowser.py]: Please revert it, I'll investigate what's happening with that.

Comment (goanpeca) [inline spyderlib/widgets/externalshell/namespacebrowser.py]: I can explain what happened... the way it was before the `import_data` was being call with a boolean as a first argument instead of the filenames like it should, when I removed the slot, the function was called as it should have

Comment (goanpeca) [inline spyderlib/widgets/externalshell/namespacebrowser.py]: I will put a FIXME with a note as well

Comment (ccordoba12) [inline spyderlib/widgets/externalshell/namespacebrowser.py]: Ok

**Comment by goanpeca** [comment]
@ccordoba12 done..
